StyleCop.Analyzers/StyleCop.Analyzers/DocumentationRules/SA1615ElementReturnValueMustBeDocumented.cs

```diff
@@ -48,7 +53,49 @@ public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics
         /// <inheritdoc/>
         public override void Initialize(AnalysisContext context)
         {
-            // TODO: Implement analysis
+            context.RegisterSyntaxNodeAction(this.HandleMethodDeclaration, SyntaxKind.MethodDeclaration);
+            context.RegisterSyntaxNodeAction(this.HandleDelegateDeclaration, SyntaxKind.DelegateDeclaration);
+        }
+
+        private void HandleMethodDeclaration(SyntaxNodeAnalysisContext context)
+        {
+            var node = context.Node as MethodDeclarationSyntax;
+
+            this.HandleDeclaration(context, node.ReturnType);
+        }
+
+        private void HandleDelegateDeclaration(SyntaxNodeAnalysisContext context)
+        {
+            var node = context.Node as DelegateDeclarationSyntax;
+
+            this.HandleDeclaration(context, node.ReturnType);
+        }
+
+        private void HandleDeclaration(SyntaxNodeAnalysisContext context, TypeSyntax returnType)
+        {
+            var predefenedType = returnType as PredefinedTypeSyntax;
+
+            if (predefenedType != null && predefenedType.Keyword.IsKind(SyntaxKind.VoidKeyword))
+            {
+                // There is no return value
+                return;
+            }
+
+            var documentationStructure = XmlCommentHelper.GetDocumentationStructure(context.Node);
+
+            if (documentationStructure != null)
+            {
+                if (XmlCommentHelper.GetTopLevelElement(documentationStructure, XmlCommentHelper.InheritdocXmlTag) != null)
+                {
+                    // Don't report if the documentation is inherited.
+                    return;
+                }
+
+                if (XmlCommentHelper.GetTopLevelElement(documentationStructure, XmlCommentHelper.ReturnsXmlTag) == null)
+                {
+                    context.ReportDiagnostic(Diagnostic.Create(Descriptor, returnType.GetLocation()));
+                }
+            }
         }
```

Comment:
I would prefer:
```C#
if (documentationStructure == null)
{
    // Don't report if there is no documentation (SA1600 should be the only diagnostic reported)
    return;
}
```
instead of the code above to clearly signal that not reporting when there is no documentation is a conscious action. 

@pdelvo What do you think?